src/libs/antares/study/area/list.cpp

```diff
@@ -894,15 +894,10 @@ static bool AreaListLoadFromFolderSingleArea(Study& study,
         {
             buffer.clear() << study.folderInput << SEP << "hydro" << SEP << "series";
             ret = hydroSeries->LoadMaxPower(area.id, buffer) && ret;
-
-            if (study.usedByTheSolver)
-            {
-                hydroSeries->EqualizeMaxPowerTSsizes(area);
-            }
-            else
-                hydroSeries->setHydroModulability(area);
         }
 
+        hydroSeries->computeTSCount();
```

Comment:
I think it would be best not to store this value, but instead to compute it every time. Fewer errors.

Reply:
Yes but we duplicate code.
Ideally simulation should not know if TS are generated or read.
This TS acquisition should be done while loading input data.
This would avoid have further complications like that.
I know that TS generation will disappear, even if we don't know when.
Anyway, this issue will be handled later, in a way or another, if it is gods will.

Reply:
Yes but it's slow and complicated. It could be more simple.
Ideally simulation should not know if TS are generated or read.
This TS acquisition should be done while loading input data.
It would allow :
- to would avoid have further complications like that.
- to probably make some more simplifications in **timeseries-numbers.cpp**

I know that TS generation will disappear, but we don't know when.
Anyway, this issue will be handled later, in a way or another. 
We should create a ticket to study the benefit we would get by taking this path.